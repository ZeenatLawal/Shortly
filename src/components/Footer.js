import React from 'react';
import logo from '../images/logo-white.svg';
import facebook from '../images/icon-facebook.svg';
import twitter from '../images/icon-twitter.svg';
import pinterest from '../images/icon-pinterest.svg';
import instagram from '../images/icon-instagram.svg';
import '../styles/Footer.css';

const Footer = () => (
  <footer className="footer">
    <div className="main d-flex flex-column align-items-center flex-lg-row align-items-lg-start">
      <img src={logo} alt="Shortly Logo" className="footer-logo" />
      <div className="d-lg-flex flex-lg-row justify-content-lg-between w-100">
        <div>
          <h6>Features</h6>
          <p>Link Shortening</p>
          <p>Branded Links</p>
          <p>Analytics</p>
        </div>
        <div>
          <h6>Resources</h6>
          <p>Blog</p>
          <p>Developers</p>
          <p>Support</p>
        </div>
        <div>
          <h6>Company</h6>
          <p>About</p>
          <p>Our Team</p>
          <p>Careers</p>
          <p>Contact</p>
        </div>
        <div className="pt-4 pt-lg-0">
          <img src={facebook} alt="Facebook icon" className="pe-4 footer-icon" />
          <img src={twitter} alt="Twitter icon" className="pe-4 footer-icon" />
          <img src={pinterest} alt="Pinterest icon" className="pe-4 footer-icon" />
          <img src={instagram} alt="Instagram icon" className="footer-icon" />
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
