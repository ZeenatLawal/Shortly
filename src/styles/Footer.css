.footer {
  background-color: var(--dark-blue);
  color: white;
  text-align: center;
  padding: 3rem 0;
}

.footer-logo {
  width: 8rem;
  height: 2rem;
  margin-bottom: 2rem;
}

.footer h6 {
  padding-top: 1rem;
  padding-bottom: 1rem;
  font-weight: 700;
}

.footer p {
  font-size: 0.8rem;
  color: var(--gray);
}

.footer p:hover {
  color: var(--cyan);
}

.footer-icon:hover {
  filter: 
    invert(0.5)
    sepia(1)
    hue-rotate(180deg)
    saturate(66)
    brightness(49);
}

@media screen and (min-width: 1024px) {
  .footer {
    text-align: left;
  }

  .footer-logo {
    margin-right: 12rem;
  }

  .footer h6 {
    padding-top: 0;
  }
}
