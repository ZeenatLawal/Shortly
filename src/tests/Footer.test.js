import renderer from 'react-test-renderer';
import Footer from '../components/Footer';

describe('Footer component', () => {
  test('Snapshot test', () => {
    const footer = renderer.create(
      <Footer />,
    )
      .toJSON();
    expect(footer).toMatchSnapshot();
  });
});
