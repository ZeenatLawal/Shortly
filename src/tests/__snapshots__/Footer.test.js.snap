// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Snapshot test 1`] = `
<footer
  className="footer"
>
  <div
    className="main d-flex flex-column align-items-center"
  >
    <img
      alt="Shortly Logo"
      className="footer-logo"
      src="logo-white.svg"
    />
    <div>
      <h6>
        Features
      </h6>
      <p>
        Link Shortening
      </p>
      <p>
        Branded Links
      </p>
      <p>
        Analytics
      </p>
    </div>
    <div>
      <h6>
        Resources
      </h6>
      <p>
        Blog
      </p>
      <p>
        Developers
      </p>
      <p>
        Support
      </p>
    </div>
    <div>
      <h6>
        Company
      </h6>
      <p>
        About
      </p>
      <p>
        Our Team
      </p>
      <p>
        Careers
      </p>
      <p>
        Contact
      </p>
    </div>
    <div
      className="pt-4 pb-5"
    >
      <img
        alt="Facebook icon"
        className="pe-4 footer-icon"
        src="icon-facebook.svg"
      />
      <img
        alt="Twitter icon"
        className="pe-4 footer-icon"
        src="icon-twitter.svg"
      />
      <img
        alt="Pinterest icon"
        className="pe-4 footer-icon"
        src="icon-pinterest.svg"
      />
      <img
        alt="Instagram icon"
        className="footer-icon"
        src="icon-instagram.svg"
      />
    </div>
  </div>
</footer>
`;
